odoo/fields.py

```diff
@@ -1171,7 +1171,7 @@ def mapped(self, records):
             # many records as possible. If not done this way, scenarios such as
             # [rec.line_ids.mapped('name') for rec in recs] would generate one
             # query per record in `recs`!
-            remaining = records._browse(records.env, records[len(vals):]._ids, records._prefetch_ids)
+            remaining = records.__class__(records.env, records[len(vals):]._ids, records._prefetch_ids)
```

Comment:
Could probably lift `records.__class__` out of the loop.

Reply:
It seems a bit too much, no ? 
Because if it is worth it, then change `records[len(vals):]._ids` into `records._ids[len(vals):]` to avoid a useless browse done by `__getitem__` will be far more useful ?